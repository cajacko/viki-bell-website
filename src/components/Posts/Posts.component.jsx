import React from 'react';
import TaxonomyTitle from 'components/TaxonomyTitle/TaxonomyTitle';
import Breadcrumbs from 'components/Breadcrumbs/Breadcrumbs';
import PostLoop from 'components/PostLoop/PostLoop.container';
import Item from 'components/Item/Item';

const Posts = ({ taxonomy, value, categoriesBySlug }) => (
  <div>
    <Item element={TaxonomyTitle} itemId={categoriesBySlug[value]} />
    <Breadcrumbs taxonomy={taxonomy} value={value} />
    <PostLoop taxonomy={taxonomy} value={value} />
  </div>
  );

export default Posts;
