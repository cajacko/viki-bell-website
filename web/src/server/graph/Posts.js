import Post from './Post';

// Function to get all posts
const Posts = () => [
  new Post(2),
  new Post(45),
  new Post(345),
  new Post(48765),
];

export default Posts;
