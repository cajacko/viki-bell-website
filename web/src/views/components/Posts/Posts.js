/* @flow */

import React from 'react';
import PropTypes from 'prop-types';
import PostsItem from 'components/PostsItem/PostsItem';

const Posts = ({ posts }) => (
  <div>
    {
      posts.map(({ id, title, date, image, category, imageAlt, slug }) => (
        <PostsItem
          key={id}
          title={title}
          date={date}
          image={image}
          category={category}
          imageAlt={imageAlt}
          slug={slug}
        />
      ))
    }
  </div>
);

Posts.propTypes = {
  posts: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Posts;
