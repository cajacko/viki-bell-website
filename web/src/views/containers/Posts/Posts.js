/* @flow */

import React from 'react';
import {
  QueryRenderer,
  graphql,
} from 'react-relay';
import PropTypes from 'prop-types';
import moment from 'moment';
import environment from 'views/relayEnvironment';
import Posts from 'components/Posts/Posts';

const query = graphql`
  query PostsQuery {
    posts(first: 5) {
      edges {
        cursor
        node {
          id
          postId
          title
          date
          image
          category
          imageAlt
          slug
        }
      }
      pageInfo {
        hasNextPage
      }
    }
  }
`;

const PostsContainer = () => (
  <QueryRenderer
    environment={environment}
    query={query}
    render={({ error, props }) => {
      if (error) {
        return <div>{error.message}</div>;
      } else if (props) {
        const posts = [];

        props.posts.edges.map((post) => {
          const finalPostObject = Object.assign({}, post.node);

          finalPostObject.date = moment.unix(finalPostObject.date);

          return posts.push(finalPostObject);
        });

        return (
          <Posts
            posts={posts}
          />
        );
      }

      return <div>Loading</div>;
    }}
  />
);

PostsContainer.propTypes = {
  posts: PropTypes.shape({
    edges: PropTypes.arrayOf(PropTypes.shape({

    })),
  }),
};

PostsContainer.defaultProps = {
  posts: {
    edges: [],
  },
};

export default PostsContainer;
