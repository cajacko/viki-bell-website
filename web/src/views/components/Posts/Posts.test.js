import React from 'react';
import moment from 'moment';
import Posts from 'components/Posts/Posts';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Posts
      posts={[
        {
          id: 234,
          title: 'Post title',
          date: moment('2016-09-09 12:12:12'),
          image: 'https://unsplash.it/400/300',
          category: 'Life',
          imageAlt: 'Image alt',
          slug: 'post-title',
        },
        {
          id: 345,
          title: 'Post title 2',
          date: moment('2016-09-04 12:12:12'),
          image: 'https://unsplash.it/400/300',
          category: 'Life',
          imageAlt: 'Image alt',
          slug: 'post-title',
        },
        {
          id: 3456,
          title: 'Post title 3',
          date: moment('2016-09-02 12:12:12'),
          image: 'https://unsplash.it/400/300',
          category: 'Life',
          imageAlt: 'Image alt',
          slug: 'post-title',
        },
      ]}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
