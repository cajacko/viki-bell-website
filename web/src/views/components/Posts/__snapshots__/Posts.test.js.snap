// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <article>
    <a
      href="post-title"
    >
      <img
        alt="Image alt"
        src="https://unsplash.it/400/300"
      />
    </a>
    <p>
      2016-09-09
    </p>
    <a
      href="post-title"
    >
      <h2>
        Post title
      </h2>
    </a>
    <a
      href="post-title"
    >
      Life
    </a>
  </article>
  <article>
    <a
      href="post-title"
    >
      <img
        alt="Image alt"
        src="https://unsplash.it/400/300"
      />
    </a>
    <p>
      2016-09-04
    </p>
    <a
      href="post-title"
    >
      <h2>
        Post title 2
      </h2>
    </a>
    <a
      href="post-title"
    >
      Life
    </a>
  </article>
  <article>
    <a
      href="post-title"
    >
      <img
        alt="Image alt"
        src="https://unsplash.it/400/300"
      />
    </a>
    <p>
      2016-09-02
    </p>
    <a
      href="post-title"
    >
      <h2>
        Post title 3
      </h2>
    </a>
    <a
      href="post-title"
    >
      Life
    </a>
  </article>
</div>
`;
